packages/router/src/router.ts

```diff
@@ -342,10 +331,10 @@ export class Router {
 
   /** @internal */
   resetRootComponentType(rootComponentType: Type<any>): void {
-    this.rootComponentType = rootComponentType;
     // TODO: vsavkin router 4.0 should make the root component set to null
     // this will simplify the lifecycle of the router.
-    this.routerState.root.component = this.rootComponentType;
+    this.routerState.root.component = rootComponentType;
+    this.navigationTransitions.rootComponentType = rootComponentType;
```

Comment:
If you guys still care about code size, I think this is just as readable:
```suggestion
    this.routerState.root.component = this.navigationTransitions.rootComponentType = rootComponentType;
```